src/diffusers/pipelines/kandinsky/pipeline_kandinsky.py

```diff
@@ -0,0 +1,388 @@
+            dtype=prompt_embeds.dtype, device=device
+        )
+
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps_tensor = self.scheduler.timesteps
+
+        num_channels_latents = self.unet.config.in_channels
+
+        height, width = get_new_h_w(height, width, self.movq_scale_factor)
+
+        # create initial latent
+        latents = self.prepare_latents(
+            (batch_size, num_channels_latents, height, width),
+            text_encoder_hidden_states.dtype,
+            device,
+            generator,
+            latents,
+            self.scheduler,
+        )
+
+        for i, t in enumerate(self.progress_bar(timesteps_tensor)):
+            # expand the latents if we are doing classifier free guidance
+            latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
+
+            added_cond_kwargs = {"text_embeds": prompt_embeds, "image_embeds": image_embeds}
+            noise_pred = self.unet(
+                sample=latent_model_input,  # [2, 4, 96, 96]
+                timestep=t,
+                encoder_hidden_states=text_encoder_hidden_states,
+                added_cond_kwargs=added_cond_kwargs,
+            ).sample
+
+            # YiYi Notes: CFG is currently implemented exactly as original repo as a baseline,
+            # i.e. we apply cfg to predicted noise, and take predicted variance as it is (uncond + cond)
+            # this means the our latent shape is batch_size *2 instad batch_size
+
+            if do_classifier_free_guidance:
+                noise_pred, variance_pred = noise_pred.split(latents.shape[1], dim=1)
+                noise_pred_uncond, noise_pred_text = noise_pred.chunk(2)
+                variance_pred_uncond, variance_pred_text = variance_pred.chunk(2)
+                noise_pred = noise_pred_uncond + guidance_scale * (noise_pred_text - noise_pred_uncond)
+                noise_pred = torch.cat([noise_pred] * 2)
+                variance_pred = torch.cat([variance_pred_uncond, variance_pred_text])
+                noise_pred = torch.cat([noise_pred, variance_pred], dim=1)
+
+            if i + 1 == timesteps_tensor.shape[0]:
+                prev_timestep = None
+            else:
+                prev_timestep = timesteps_tensor[i + 1]
+
+            # compute the previous noisy sample x_t -> x_t-1
+            latents = self.scheduler.step(
+                noise_pred,
+                t,
+                latent_model_input,
+                prev_timestep=prev_timestep,
+                generator=generator,
+                batch_size=batch_size,
+            ).prev_sample
+
+            _, latents = latents.chunk(2)
+
+        # post-processing
+        image = self.movq.decode(latents, force_not_quantize=True)["sample"]
+
+        image = image * 0.5 + 0.5
+        image = image.clamp(0, 1)
+        image = image.cpu().permute(0, 2, 3, 1).float().numpy()
+
+        if output_type == "pil":
```

Comment:
Let's make sure to allow all `output_type == "pil"` or `output_type == "pt"` or `output_type == "np"` , but only those (so let's add ability to return PT and also add a test that checks that only pt np and pil are allowed)